pkg/trtl/peers.go

```diff
@@ -2,24 +2,45 @@ package trtl
 
 import (
 	context "context"
+	"errors"
 
 	"github.com/rs/zerolog/log"
-	"github.com/trisacrypto/directory/pkg/gds/store"
+	"github.com/trisacrypto/directory/pkg/gds/store/iterator"
 	"github.com/trisacrypto/directory/pkg/trtl/peers/v1"
 	codes "google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
 )
 
+type TemporaryPeerStore interface {
+	CreatePeer(peer *peers.Peer) (string, error)
+	DeletePeer(id string) error
+	ListPeers() iterator.Iterator
+	AllPeers() ([]*peers.Peer, error)
+}
+
+type notImplementedStore struct{}
+
+func (s *notImplementedStore) CreatePeer(peer *peers.Peer) (string, error) {
+	return "", errors.New("not implemented")
+}
+func (s *notImplementedStore) DeletePeer(id string) error   { return errors.New("not implemented") }
+func (s *notImplementedStore) ListPeers() iterator.Iterator { return nil }
+func (s *notImplementedStore) AllPeers() ([]*peers.Peer, error) {
+	return nil, errors.New("not implemented")
+}
+
```

Comment:
Sorry, you went through all that work to create the store, but this is what I thought you had in mind for a temporary store. I didn't want to add too much redirection by having interfaces on top of interfaces. Let's create a story to move the Peer store code to this file so it can interact with the database directly.